lms/djangoapps/discussion_api/tests/test_api.py

```diff
@@ -573,3 +572,368 @@ def test_pagination(self):
         self.register_get_threads_response([], page=3, num_pages=3)
         with self.assertRaises(Http404):
             get_thread_list(self.request, self.course.id, page=4, page_size=10)
+
+
+@ddt.ddt
+class GetCommentListTest(CommentsServiceMockMixin, ModuleStoreTestCase):
+    """Test for get_thread_list"""
+    def setUp(self):
+        super(GetCommentListTest, self).setUp()
+        httpretty.reset()
+        httpretty.enable()
+        self.addCleanup(httpretty.disable)
+        self.maxDiff = None  # pylint: disable=invalid-name
+        self.user = UserFactory.create()
+        self.register_get_user_response(self.user)
+        self.request = RequestFactory().get("/test_path")
+        self.request.user = self.user
+        self.course = CourseFactory.create()
+        CourseEnrollmentFactory.create(user=self.user, course_id=self.course.id)
+        self.author = UserFactory.create()
+
+    def make_minimal_cs_thread(self, overrides=None):
+        """
+        Create a thread with the given overrides, plus the course_id if not
+        already in overrides.
+        """
+        overrides = dict(overrides or {})
+        overrides.setdefault("course_id", unicode(self.course.id))
+        return make_minimal_cs_thread(overrides)
+
+    def get_comment_list(self, thread, endorsed=None, page=1, page_size=1):
```

Comment:
This helper function name was a bit confusing for me. It seems like there are two things going on in this function, register a thread and get a comment list, where get_comment_list is an already existing method. Only suggestion I can think of is "register_thread_and_get_comment_list" which seems pretty long.